### PR TITLE
feat(rulesets): add support for PR labeled events

### PR DESCRIPTION
**Pull request by wsan3**
## Description
Request for Vela to process and create builds for pull request [labeled](https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=labeled#pull_request) and [unlabeled](https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=unlabeled#pull_request) webhooks.

## Related
GitHub [issue #913](https://github.com/go-vela/community/issues/913)
GitHub [ideas #768](https://git.target.com/orgs/vela/projects/36/views/1?filterQuery=win+san&pane=issue&itemId=19205)

### CODE REVIEW

File: library/actions/pull.go [hunk 1]
```diff
@@ -91,6 +97,17 @@ func (a *Pull) GetReopened() bool {
 	return *a.Reopened
 }
 
+// GetLabeled returns the Labeled field from the provided Pull. If the object is nil,
+// or the field within the object is nil, it returns the zero value instead.
+func (a *Pull) GetLabeled() bool {
+	// return zero value if Pull type or Reopened field is nil
```
Comment (ecrupper) [inline library/actions/pull.go:103]:
```suggestion
	// return zero value if Pull type or Labeled field is nil
```

**Review by ecrupper — CHANGES_REQUESTED** [review]
Looks great. One typo I noticed and that's it

File: pipeline/ruleset.go [hunk 2]
```diff
@@ -57,6 +58,7 @@ type (
 		Status   string   `json:"status,omitempty"  yaml:"status,omitempty"`
 		Tag      string   `json:"tag,omitempty"     yaml:"tag,omitempty"`
 		Target   string   `json:"target,omitempty"  yaml:"target,omitempty"`
+		Label    string   `json:"label,omitempty"   yaml:"label,omitempty"`
```
Comment (ecrupper) [inline pipeline/ruleset.go:61]: Thoughts on extending this rule to include other `pull_request` events, which would result in passing in a slice of label strings? So if a PR has labels `bug` and `v2`, we'd pass in a label set of `[]string{"bug", "v2"}`?

File: pipeline/ruleset.go [hunk 3]
```diff
@@ -111,160 +113,159 @@ func (r *Rules) Empty() bool {
...
+	}
+
+	// iterate through each pattern in the ruletype
+	for _, pattern := range *r {
+		for _, value := range data {
+			match, err := match(value, matcher, pattern)
+			if err != nil {
+				return false, err
+			}
+			if match {
```
[the remark this answers is not in the document]
Comment (plyr4) [reply pipeline/ruleset.go:233]: @wsan3 can we fix this linter warning? 
```suggestion

			if match {
```

**Review by plyr4 — CHANGES_REQUESTED** [review]

**Review by plyr4 — APPROVED** [review]

**Review by wass3rw3rk — APPROVED** [review]

**Pull request merged**